streams/src/main/java/org/apache/kafka/streams/state/internals/AbstractRocksDBSegmentedBytesStore.java

```diff
@@ -216,6 +226,11 @@ public void put(final Bytes key,
             expiredRecordSensor.record(1.0d, ProcessorContextUtils.currentSystemTime(context));
             LOG.warn("Skipping record for expired segment.");
         } else {
+            final InternalProcessorContext internalContext = asInternalProcessorContext(context);
```

Comment:
Until the state store gets upgraded to use the `StateStoreContext`, we need to do this if we want to record the position on put